### ISSUE TITLE
bug: Higher memory usage after upgrading to 2.2.0

### ISSUE THREAD

**Issue by suzaku** [localstack/localstack#8837]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Start observing "ran out of memory" errors after upgrading from 1.4.0 to 2.2.0 on Aug 03.

<img width="1134" alt="Screenshot 2023-08-07 at 10 53 13" src="https://github.com/localstack/localstack/assets/116473/de0a1422-0992-417a-9c4a-6b38275be017">

<img width="760" alt="Screenshot 2023-08-07 at 10 53 39" src="https://github.com/localstack/localstack/assets/116473/0161d866-db40-4660-9497-d11d80b68289">

### Expected Behavior

According to #8269, we can set `SQS_DISABLE_CLOUDWATCH_METRICS` to `"1"` to disable this memory-consuming feature completely since 2.0.

This is important to us because we do use a lot of SQS queues in `localstack`.

When we upgraded from 1.2 to 1.4, I expected the memory consumption to go up because this environment variable would not work, but I didn't see any growth in the number of OOM errors.
When we later upgraded from 1.4 to 2.0, I expected the memory usage to stay stable, but instead it went up and caused lots of OOMs.

Besides SQS, we also use s3 and kinesis.

### How are you starting LocalStack?

Custom (please describe below)

#### How are you starting localstack (e.g., `bin/localstack` command, arguments, or `docker-compose.yml`)

    Run localstack/localstack:2.2.0 in Kubernetes.

    Environment variables:

```bash
      SERVICES:                        sqs,kinesis,dynamodb,cloudwatch,s3
      EAGER_SERVICE_LOADING:           1
      FORCE_NONINTERACTIVE:            true
      KINESIS_SHARD_LIMIT:             250
      KINESIS_LATENCY:                 0
      LOCALSTACK_HOSTNAME:             localstack
      LAMBDA_EXECUTOR:                 local
      FILESYSTEM_ROOT:                 /
      SQS_DISABLE_CLOUDWATCH_METRICS:  1
```

### Environment

```markdown
- OS: Linux
- LocalStack: 2.2.0
```

**Comment by thrau** [comment]
hi @suzaku, could you check if you have these issues with the latest image? the issue should be resolved with #9134

**Comment by suzaku** [comment]
> hi @suzaku, could you check if you have these issues with the latest image? the issue should be resolved with #9134

I will give it a try next week, thanks.
